pallets/did/src/lib.rs

```diff
@@ -114,87 +116,177 @@ pub mod pallet {
 	}
 
 	#[pallet::error]
-	pub enum Error<T> {}
+	pub enum Error<T> {
+		CredentialAlreadyAdded,
+		CredentialDoesNotExist,
+		IssuerAlreadyExists,
+		IssuerDoesNotExist,
+		IssuerNotActive,
+		IssuerNotRevoked,
+		MaxCredentials,
+	}
 
 	#[pallet::call]
 	impl<T: Config> Pallet<T> {
-
+		#[pallet::call_index(0)]
 		#[pallet::weight(1000000)]
 		pub fn create_did(
 			origin: OriginFor<T>,
 			authenticator: H160, // Ethereum Address
-			signature: DidSignature
+			signature: DidSignature,
 		) -> DispatchResult {
 			let controller = ensure_signed(origin)?;
-			Self::deposit_event(Event::DidCreated{
-				controller,
-				authenticator,
-				signature
-			});
+			Self::deposit_event(Event::DidCreated { controller, authenticator, signature });
 			Ok(())
 		}
 
+		#[pallet::call_index(1)]
 		#[pallet::weight(1000000)]
 		pub fn add_issuer(origin: OriginFor<T>, issuer: DidIdentifierOf<T>) -> DispatchResult {
 			// Origin ONLY GovernanceOrigin
 			T::GovernanceOrigin::ensure_origin(origin)?;
-			// Add issuer to database with status Active
+			ensure!(!Issuers::<T>::contains_key(&issuer), Error::<T>::IssuerAlreadyExists);
+
+			Issuers::<T>::try_mutate(issuer.clone(), |maybe_info| -> DispatchResult {
+				*maybe_info = Some(IssuerInfo { status: IssuerStatus::Active });
+				Self::deposit_event(Event::IssuerStatusActive { issuer });
+				Ok(())
+			})?;
 			Ok(())
 		}
 
+		#[pallet::call_index(2)]
 		#[pallet::weight(1000000)]
 		pub fn revoke_issuer(origin: OriginFor<T>, issuer: DidIdentifierOf<T>) -> DispatchResult {
 			// Origin ONLY GovernanceOrigin
 			T::GovernanceOrigin::ensure_origin(origin)?;
+
 			// Change status to Revoked
+			Issuers::<T>::try_mutate_exists(issuer.clone(), |maybe_info| -> DispatchResult {
+				let mut info = maybe_info.as_mut().ok_or(Error::<T>::IssuerDoesNotExist)?;
+				ensure!(
+					*info == IssuerInfo { status: IssuerStatus::Active },
+					Error::<T>::IssuerNotActive
+				);
+				*info = IssuerInfo { status: IssuerStatus::Revoked };
+				Self::deposit_event(Event::IssuerStatusRevoked { issuer });
+				Ok(())
+			})?;
 			Ok(())
 		}
 
+		#[pallet::call_index(3)]
+		#[pallet::weight(1000000)]
+		pub fn reactivate_issuer(
+			origin: OriginFor<T>,
+			issuer: DidIdentifierOf<T>,
+		) -> DispatchResult {
+			// Origin ONLY GovernanceOrigin
+			T::GovernanceOrigin::ensure_origin(origin)?;
+
+			// Change status to Active
+			Issuers::<T>::try_mutate_exists(issuer.clone(), |maybe_info| -> DispatchResult {
+				let mut info = maybe_info.as_mut().ok_or(Error::<T>::IssuerDoesNotExist)?;
+				ensure!(
+					*info == IssuerInfo { status: IssuerStatus::Revoked },
+					Error::<T>::IssuerNotRevoked
+				);
+				*info = IssuerInfo { status: IssuerStatus::Active };
+				Self::deposit_event(Event::IssuerStatusReactived { issuer });
+				Ok(())
+			})?;
+			Ok(())
+		}
+
+		#[pallet::call_index(4)]
 		#[pallet::weight(1000000)]
 		pub fn delete_issuer(origin: OriginFor<T>, issuer: DidIdentifierOf<T>) -> DispatchResult {
```

Comment:
```suggestion
		pub fn remove_issuer(origin: OriginFor<T>, issuer: DidIdentifierOf<T>) -> DispatchResult {
```
Just to be consistent with the rest of the extrinsics naming